src/jit/lower.cpp

```diff
@@ -1330,9 +1328,9 @@ void Lowering::LowerArg(GenTreeCall* call, GenTree** ppArg)
             GenTree* argLo = arg->gtGetOp1();
             GenTree* argHi = arg->gtGetOp2();
 
-            GenTreeFieldList* fieldList = new (comp, GT_FIELD_LIST) GenTreeFieldList(argLo, 0, TYP_INT, nullptr);
-            // Only the first fieldList node (GTF_FIELD_LIST_HEAD) is in the instruction sequence.
-            (void)new (comp, GT_FIELD_LIST) GenTreeFieldList(argHi, 4, TYP_INT, fieldList);
+            GenTreeFieldList* fieldList = new (comp, GT_FIELD_LIST) GenTreeFieldList();
+            fieldList->gtUses           = new (comp, CMK_ASTNode) GenTreeFieldList::Use(argLo, 0, TYP_INT);
+            fieldList->gtUses->SetNext(new (comp, CMK_ASTNode) GenTreeFieldList::Use(argHi, 4, TYP_INT));
```

Comment:
Yes, every time I did one of these "stop using `GT_LIST`" changes. The problem is that it's kind of risky to have something like "AddUse" because we only keep track of the head of the linked list and appending would require traversing the list to find the tail. I think there are 2 ways to handle this:
* add something similar to `gtNewCallArgs` - `gtNewFieldList` or something like that - that hides away this manual list manipulation in those cases where the number of fields is statically known
* add something like `UseListBuilder` - a small class that keeps track of both the head and tail and can be used in the rest of the cases.

This should cover almost all cases. AFAIR the only place where we still need to do some manual linked list manipulation in long decomposition (and that's the main reason I maintained the existing linked list approach, otherwise arrays would be a very tempting alternative).

Reply:
Right. Another option would be to have constructor overloads for the known-number-of-fields cases. But I'm OK with deferring consideration of such a change.

Reply:
Though in the case of field list (and PHIs) I don't think there's anything stopping us from adding the tail of the list to the node, there's enough space in `GenTreeFieldList` for that. It's only calls that are problematic, they're large enough even without adding more pointers to them.

Reply:
Well, I added `AddField` and `InsertField`. Not sure I like all this too much, in part due to the side effect inheritance. Sure, `GenTreeArgList` did this automatically (and so do other `GenTree` derived classes). And I now realize that this happens even in LIR, where side effects are not supposed to be inherited. So I also added `AddFieldLIR` and `InsertFieldLIR` to deal with this case.

Opinions?

@sandreenko too